scripts/imgtool/image.py

```diff
@@ -176,6 +176,12 @@ def save(self, path):
                 raise Exception("Input file does not provide a base address")
             h = IntelHex()
             h.frombytes(bytes=self.payload, offset=self.base_addr)
+            if hex_addr is not None:
```

Comment:
Is it possible to do this with h.segments(), h.gets(), and using h.puts() on a new IntelHex object?

Reply:
Oh, I completely missed those methods! Fixed, thanks.